panel/tests/test_echarts.py

```diff
@@ -0,0 +1,70 @@
+# Minimal echarts test server, start with:
+# panel serve panel/tests/test_echarts.py --allow-websocket-origin=localhost:5005 --port=5005
+
+import time
+import pandas as pd
+import panel as pn
+
+COLOR = "#E1477E"
+
+data_tree = [{'name':'ROOT', 'children': [{'name': 'Child Non-clickable', 'value': 1}, {'name': 'Child Clickable', 'value': 2, 'id': 'ABC'}]}]
+
+# Spinner styles: 'arc', 'arcs', 'bar', 'dots', 'petal'
+pn.extension(loading_spinner='arc', loading_color=COLOR)
+
+btn = pn.widgets.Button(name="Refresh", button_type="primary")
+
+echart_option = {
+    "tooltip": {
+        "trigger": 'item',
+        "triggerOn": 'mousemove'
+    },
+    "series": [
+        {
+            "type": 'tree',
+            "data": data_tree,
+            "top": '18%',
+            "bottom": '14%',
+            "layout": 'radial',
+            "symbol": 'emptyCircle',
+            "symbolSize": 10,
+            "initialTreeDepth": 2,
+            "animationDurationUpdate": 750,
+            "emphasis": {
+                "focus": 'descendant'
+            }
+        }
+    ],
+    "responsive": True
+}
+
+event_config = {
+    # 'click': 'series.tree', # Tests for Echarts event names with query
+    # 'click': None, # Tests for Echarts event names without query
+    # 'click': {'name': 'Child Clickable'}, # Tests for Echarts event names without query
+    # 'click': {'query': 'series.tree', 'base_url': 'https://www.TEST.de/AssetDetail.aspx?AssetId=',
+    #     'identifier': 'id'}, # Tests for new browser tab Echarts event
+    'click': {'query': 'series.tree', 'handler': 'e => console.log("I got an event:", e)'}, # Tests for handler Echarts event
```

Comment:
The js `handler` is very clever and useful. It is something that could be valuable in general every time a JS library is wrapped in Panel. It would be nice to have a common "api" or naming convention for that. So I would suggest @philippjfr also gives his comments and suggestions on this.

And now you use it to handle events. In my implementation of `panel-highcharts` I have a convention for specifying js functions as the values of the json/ dict that constructs the chart. That convention is that a value that looks like `'function ( ) { SCRIPT }'` is a javascript function and will be converted to a such before being passed to the constructor of the chart. A similar cross panel convention would be nice to have.

Reply:
+1 and as you guys know much better how this fits in best to the overall logic and convention happy to implement as soon you agreed :-)